packages/material-ui-system/src/breakpoints.js

```diff
@@ -36,7 +36,13 @@ export function handleBreakpoints(props, propValue, styleFromPropValue) {
   if (typeof propValue === 'object') {
     const themeBreakpoints = props.theme.breakpoints || defaultBreakpoints;
     return Object.keys(propValue).reduce((acc, breakpoint) => {
-      acc[themeBreakpoints.up(breakpoint)] = styleFromPropValue(propValue[breakpoint]);
+      // key is breakpoint
+      if(defaultBreakpoints.keys.indexOf(breakpoint) != -1) {
```

Comment:
Now that we use this in object of styles, where we can have pseudo styles, selectors etc, we need to know when some key is valid breakpoint. If the key is not a valid breakpoint, than we are returning the initial value. @oliviertassinari does this change makes sense? Should we maybe create separate functionality for the `sx` prop rather then to reuse the same?

Reply:
I was hoping we could drop the `.keys` part of the theme to rely on `Object.keys(theme.breakpoints.values)` on v5. I wonder if that could make customization of the theme simpler.

Also, why use the default theme over the current one?

The motivation for the change makes sense. No objections to put the logic here. I think that we should add a test for it with the sx prop. The breakpoint logic has a couple of bugs that may need a refacing.

Reply:
Replaced with `themeBreakpoints` instead of `defaultBreakpoints`, that was mistake on my end. I can replace it with `Object.keys(theme.breakpoints.values)`. Let me first introduce tests for it, to make sure it still works :)